<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>rsnps tutorial</title>
<style type="text/css">
/**
 * Prism.s theme ported from highlight.js's xcode style
 */
pre code {
  padding: 1em;
}
.token.comment {
  color: #007400;
}
.token.punctuation {
  color: #999;
}
.token.tag,
.token.selector {
  color: #aa0d91;
}
.token.boolean,
.token.number,
.token.constant,
.token.symbol {
  color: #1c00cf;
}
.token.property,
.token.attr-name,
.token.string,
.token.char,
.token.builtin {
  color: #c41a16;
}
.token.inserted {
  background-color: #ccffd8;
}
.token.deleted {
  background-color: #ffebe9;
}
.token.operator,
.token.entity,
.token.url,
.language-css .token.string,
.style .token.string {
  color: #9a6e3a;
}
.token.atrule,
.token.attr-value,
.token.keyword {
  color: #836c28;
}
.token.function,
.token.class-name {
  color: #DD4A68;
}
.token.regex,
.token.important,
.token.variable {
  color: #5c2699;
}
.token.important,
.token.bold {
  font-weight: bold;
}
.token.italic {
  font-style: italic;
}
</style>
<style type="text/css">
body {
  font-family: sans-serif;
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 1.5;
  box-sizing: border-box;
}
body, .footnotes, code { font-size: .9em; }
li li { font-size: .95em; }
*, *:before, *:after {
  box-sizing: inherit;
}
pre, img { max-width: 100%; }
pre, pre:hover {
  white-space: pre-wrap;
  word-break: break-all;
}
pre code {
  display: block;
  overflow-x: auto;
}
code { font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace; }
:not(pre) > code, code[class] { background-color: #F8F8F8; }
code.language-undefined, pre > code:not([class]) {
  background-color: inherit;
  border: 1px solid #eee;
}
table {
  margin: auto;
  border-top: 1px solid #666;
}
table thead th { border-bottom: 1px solid #ddd; }
th, td { padding: 5px; }
thead, tfoot, tr:nth-child(even) { background: #eee; }
blockquote {
  color: #666;
  margin: 0;
  padding-left: 1em;
  border-left: 0.5em solid #eee;
}
hr, .footnotes::before { border: 1px dashed #ddd; }
.frontmatter { text-align: center; }
#TOC .numbered li { list-style: none; }
#TOC .numbered { padding-left: 0; }
#TOC .numbered ul { padding-left: 1em; }
table, .body h2 { border-bottom: 1px solid #666; }
.body .appendix, .appendix ~ h2 { border-bottom-style: dashed; }
.footnote-ref a::before { content: "["; }
.footnote-ref a::after { content: "]"; }
.footnotes::before {
  content: "";
  display: block;
  max-width: 20em;
}

@media print {
  body {
    font-size: 12pt;
    max-width: 100%;
  }
  tr, img { page-break-inside: avoid; }
}
@media only screen and (min-width: 992px) {
  pre { white-space: pre; }
}
</style>
</head>
<body>
<div class="include-before">
</div>
<div class="frontmatter">
<div class="title"><h1>rsnps tutorial</h1></div>
<div class="author"><h2></h2></div>
<div class="date"><h3></h3></div>
</div>
<div class="body">
<!--
%\VignetteEngine{knitr::knitr}
%\VignetteIndexEntry{rsnps tutorial}
%\VignetteEncoding{UTF-8}
-->
<h1 id="rsnps-tutorial">rsnps tutorial</h1>
<h2 id="install-and-load-library">Install and load library</h2>
<p>When available on CRAN</p>
<pre><code class="language-r">install.packages(&quot;rsnps&quot;)
</code></pre>
<p>Or get from Github</p>
<pre><code class="language-r">install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;ropensci/rsnps&quot;)
</code></pre>
<pre><code class="language-r">library(rsnps)
</code></pre>
<h2 id="opensnp-data">OpenSNP data</h2>
<h3 id="all-genotypes">All Genotypes</h3>
<p>Get genotype data for all users at a particular SNP from <a href="https://opensnp.org">OpenSNP</a>:</p>
<pre><code class="language-r">x &lt;- allgensnp(snp='rs7412')
head(x)
#&gt;     name chromosome position           user_name   id genotype_id local_genotype
#&gt; 1 rs7412         19 44908822        R.M. Holston   22           8             CC
#&gt; 2 rs7412         19 44908822 Charles G. Sullivan 5326        3834             CC
#&gt; 3 rs7412         19 44908822   Glenn Allen Nolen   19           7             CC
#&gt; 4 rs7412         19 44908822        Angel Harris  495         223             CC
#&gt; 5 rs7412         19 44908822           Mom to AG  387         173             CC
#&gt; 6 rs7412         19 44908822            kevinmcc  285         118             CC
</code></pre>
<h3 id="all-phenotypes">All Phenotypes</h3>
<p>Get all phenotypes, their variations, and how many users have data available for a given phenotype</p>
<p>Get all data</p>
<pre><code class="language-r">x &lt;- allphenotypes(df = TRUE)
head(x)
#&gt;   id characteristic known_variations number_of_users
#&gt; 1  1      Eye color            Brown            1771
#&gt; 2  1      Eye color      Brown-green            1771
#&gt; 3  1      Eye color       Blue-green            1771
#&gt; 4  1      Eye color        Blue-grey            1771
#&gt; 5  1      Eye color            Green            1771
#&gt; 6  1      Eye color             Blue            1771
</code></pre>
<p>Output a list, then call the characteristic of interest by ‘id’ or ‘characteristic’</p>
<pre><code class="language-r">datalist &lt;- allphenotypes()
</code></pre>
<p>Get a list of all characteristics you can call</p>
<pre><code class="language-r">names(datalist)[1:10]
#&gt;  [1] &quot;Eye color&quot;                        &quot;Lactose intolerance&quot;              &quot;Handedness&quot;                       &quot;white skin&quot;                      
#&gt;  [5] &quot;Ability to find a bug in openSNP&quot; &quot;Beard Color&quot;                      &quot;Hair Color&quot;                       &quot;Ability to Tan&quot;                  
#&gt;  [9] &quot;Height&quot;                           &quot;Hair Type&quot;
</code></pre>
<p>Get data.frame for <em>ADHD</em></p>
<pre><code class="language-r">datalist[[&quot;ADHD&quot;]]
#&gt;    id characteristic                                                                                             known_variations number_of_users
#&gt; 1  29           ADHD                                                                                                        False             353
#&gt; 2  29           ADHD                                                                                                         True             353
#&gt; 3  29           ADHD                                                                               Undiagnosed, but probably true             353
#&gt; 4  29           ADHD                                                                                                           No             353
#&gt; 5  29           ADHD                                                                                                          Yes             353
#&gt; 6  29           ADHD                                                                                                Not diagnosed             353
#&gt; 7  29           ADHD                                                                  Diagnosed as not having but with some signs             353
#&gt; 8  29           ADHD                                                                                                  Mthfr c677t             353
#&gt; 9  29           ADHD                                                                                                    Rs1801260             353
#&gt; 10 29           ADHD                                                                                                  Adult onset             353
#&gt; 11 29           ADHD                                                                   Diagnosed as &quot;other hyperkinetic disorder&quot;             353
#&gt; 12 29           ADHD                                                                                 Blonde, european, green eyes             353
#&gt; 13 29           ADHD                                                                                                      Extreme             353
#&gt; 14 29           ADHD Diagnosed as hyperactive type, though it is my belief that adhd is simply a normal trait such as eye color.              353
#&gt; 15 29           ADHD                                                                                                Combined type             353
</code></pre>
<p>Get data.frame for <em>mouth size</em> and <em>SAT Writing</em></p>
<pre><code class="language-r">datalist[c(&quot;mouth size&quot;,&quot;SAT Writing&quot;)]
#&gt; $`mouth size`
#&gt;    id characteristic     known_variations number_of_users
#&gt; 1 120     mouth size               Medium             206
#&gt; 2 120     mouth size                Small             206
#&gt; 3 120     mouth size                Large             206
#&gt; 4 120     mouth size Slightly wide mouth              206
#&gt; 
#&gt; $`SAT Writing`
#&gt;    id characteristic                                        known_variations number_of_users
#&gt; 1  41    SAT Writing                                                     750             112
#&gt; 2  41    SAT Writing                                      Tested before 2005             112
#&gt; 3  41    SAT Writing                                                     800             112
#&gt; 4  41    SAT Writing                                     Country with no sat             112
#&gt; 5  41    SAT Writing                                                     N/a             112
#&gt; 6  41    SAT Writing                                 Never &amp; have ba &amp; above             112
#&gt; 7  41    SAT Writing                                                     720             112
#&gt; 8  41    SAT Writing                         Did well - don't remember score             112
#&gt; 9  41    SAT Writing                                                     511             112
#&gt; 10 41    SAT Writing                                                     760             112
#&gt; 11 41    SAT Writing                                                     780             112
#&gt; 12 41    SAT Writing                                                     700             112
#&gt; 13 41    SAT Writing Not part of sat when i took test in august 1967 at uiuc             112
#&gt; 14 41    SAT Writing                                 Not part of sat in 1961             112
#&gt; 15 41    SAT Writing                                                     620             112
#&gt; 16 41    SAT Writing                                                     560             112
</code></pre>
<h3 id="annotations">Annotations</h3>
<p>Get just the metadata</p>
<pre><code class="language-r">annotations(snp = 'rs7903146', output = 'metadata')
#&gt;          .id        V1
#&gt; 1       name rs7903146
#&gt; 2 chromosome        10
#&gt; 3   position 112998590
</code></pre>
<p>Just from PLOS journals</p>
<pre><code class="language-r">annotations(snp = 'rs7903146', output = 'plos')[c(1:2),]
#&gt;                   author
#&gt; 1        Maggie C. Y. Ng
#&gt; 2 André Gustavo P. Sousa
#&gt;                                                                                                                                      title
#&gt; 1 Meta-Analysis of Genome-Wide Association Studies in African Americans Provides Insights into the Genetic Architecture of Type 2 Diabetes
#&gt; 2                                  Genetic Variants of Diabetes Risk and Incident Cardiovascular Events in Chronic Coronary Artery Disease
#&gt;           publication_date number_of_readers                                          url                          doi
#&gt; 1 2014-08-07T00:00:00.000Z             11650 https://doi.org/10.1371/journal.pgen.1004517 10.1371/journal.pgen.1004517
#&gt; 2 2011-01-20T00:00:00.000Z              2482 https://doi.org/10.1371/journal.pone.0016341 10.1371/journal.pone.0016341
</code></pre>
<p>Just from SNPedia</p>
<pre><code class="language-r">annotations(snp = 'rs7903146', output = 'snpedia')
#&gt;                                               url                                                          summary
#&gt; 1 http://www.snpedia.com/index.php/Rs7903146(C;C) Normal (lower) risk of Type 2 Diabetes and Gestational Diabetes.
#&gt; 2 http://www.snpedia.com/index.php/Rs7903146(C;T)     1.4x increased risk for diabetes (and perhaps colon cancer).
#&gt; 3 http://www.snpedia.com/index.php/Rs7903146(T;T)                            2x increased risk for Type-2 diabetes
</code></pre>
<p>Get all annotations</p>
<pre><code class="language-r">annotations(snp = 'rs7903146', output = 'all')[1:5,]
#&gt;        .id              author
#&gt; 1 mendeley           T E Meyer
#&gt; 2 mendeley      Camilla Cervin
#&gt; 3 mendeley Nicholette D Palmer
#&gt; 4 mendeley      Ashis K Mondal
#&gt; 5 mendeley        Julian Munoz
#&gt;                                                                                                                                title publication_year
#&gt; 1                                                Diabetes genes and prostate cancer in the Atherosclerosis Risk in Communities study             2010
#&gt; 2                                                        Diabetes in Adults , Type 1 Diabetes , and Type 2 Diabetes GENETICS OF LADA             2008
#&gt; 3                                Association of TCF7L2 gene polymorphisms with reduced acute insulin response in Hispanic Americans.             2008
#&gt; 4                  Genotype and tissue-specific effects on alternative splicing of the transcription factor 7-like 2 gene in humans.             2010
#&gt; 5 Polymorphism in the transcription factor 7-like 2 (TCF7L2) gene is associated with reduced insulin secretion in nondiabetic women.             2006
#&gt;   number_of_readers open_access
#&gt; 1                 3        TRUE
#&gt; 2                 2       FALSE
#&gt; 3                 8       FALSE
#&gt; 4                13        TRUE
#&gt; 5                10        TRUE
#&gt;                                                                                                                                      url
#&gt; 1                              http://www.mendeley.com/research/diabetes-genes-prostate-cancer-atherosclerosis-risk-communities-study-4/
#&gt; 2                                        http://www.mendeley.com/research/diabetes-adults-type-1-diabetes-type-2-diabetes-genetics-lada/
#&gt; 3              http://www.mendeley.com/research/association-tcf7l2-gene-polymorphisms-reduced-acute-insulin-response-hispanic-americans/
#&gt; 4        http://www.mendeley.com/research/genotype-tissuespecific-effects-alternative-splicing-transcription-factor-7like-2-gene-humans/
#&gt; 5 http://www.mendeley.com/research/polymorphism-transcription-factor-7like-2-tcf7l2-gene-associated-reduced-insulin-secretion-nondiabet/
#&gt;                                              doi publication_date summary first_author pubmed_link journal trait pvalue pvalue_description
#&gt; 1 19/2/558 [pii]\\r10.1158/1055-9965.EPI-09-0902             &lt;NA&gt;    &lt;NA&gt;         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA               &lt;NA&gt;
#&gt; 2                         10.2337/db07-0299.Leif             &lt;NA&gt;    &lt;NA&gt;         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA               &lt;NA&gt;
#&gt; 3                           10.1210/jc.2007-1225             &lt;NA&gt;    &lt;NA&gt;         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA               &lt;NA&gt;
#&gt; 4                           10.1210/jc.2009-2064             &lt;NA&gt;    &lt;NA&gt;         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA               &lt;NA&gt;
#&gt; 5                              10.2337/db06-0574             &lt;NA&gt;    &lt;NA&gt;         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA               &lt;NA&gt;
#&gt;   confidence_interval
#&gt; 1                &lt;NA&gt;
#&gt; 2                &lt;NA&gt;
#&gt; 3                &lt;NA&gt;
#&gt; 4                &lt;NA&gt;
#&gt; 5                &lt;NA&gt;
</code></pre>
<h3 id="download">Download</h3>
<p>Download genotype data for a user from 23andme or other repo. (not evaluated in this example)</p>
<pre><code class="language-r">data &lt;- users(df=TRUE)
head(data[[1]])
fetch_genotypes(url = data[[1]][1,&quot;genotypes.download_url&quot;], rows=15)
</code></pre>
<h3 id="genotype-user-data">Genotype user data</h3>
<p>Genotype data for one or multiple users</p>
<pre><code class="language-r">genotypes(snp='rs9939609', userid=1)
#&gt; $snp
#&gt; $snp$name
#&gt; [1] &quot;rs9939609&quot;
#&gt; 
#&gt; $snp$chromosome
#&gt; [1] &quot;16&quot;
#&gt; 
#&gt; $snp$position
#&gt; [1] &quot;53786615&quot;
#&gt; 
#&gt; 
#&gt; $user
#&gt; $user$name
#&gt; [1] &quot;Bastian Greshake Tzovaras&quot;
#&gt; 
#&gt; $user$id
#&gt; [1] 1
#&gt; 
#&gt; $user$genotypes
#&gt; $user$genotypes[[1]]
#&gt; $user$genotypes[[1]]$genotype_id
#&gt; [1] 9
#&gt; 
#&gt; $user$genotypes[[1]]$local_genotype
#&gt; [1] &quot;AT&quot;
</code></pre>
<pre><code class="language-r">genotypes('rs9939609', userid='1,6,8', df=TRUE)
#&gt;    snp_name snp_chromosome snp_position                 user_name user_id genotype_id genotype
#&gt; 1 rs9939609             16     53786615 Bastian Greshake Tzovaras       1           9       AT
#&gt; 2 rs9939609             16     53786615              Nash Parovoz       6           5       AT
#&gt; 3 rs9939609             16     53786615         Samantha B. Clark       8           2       TT
</code></pre>
<pre><code class="language-r">genotypes('rs9939609', userid='1-2', df=FALSE)
#&gt; [[1]]
#&gt; [[1]]$snp
#&gt; [[1]]$snp$name
#&gt; [1] &quot;rs9939609&quot;
#&gt; 
#&gt; [[1]]$snp$chromosome
#&gt; [1] &quot;16&quot;
#&gt; 
#&gt; [[1]]$snp$position
#&gt; [1] &quot;53786615&quot;
#&gt; 
#&gt; 
#&gt; [[1]]$user
#&gt; [[1]]$user$name
#&gt; [1] &quot;Bastian Greshake Tzovaras&quot;
#&gt; 
#&gt; [[1]]$user$id
#&gt; [1] 1
#&gt; 
#&gt; [[1]]$user$genotypes
#&gt; [[1]]$user$genotypes[[1]]
#&gt; [[1]]$user$genotypes[[1]]$genotype_id
#&gt; [1] 9
#&gt; 
#&gt; [[1]]$user$genotypes[[1]]$local_genotype
#&gt; [1] &quot;AT&quot;
#&gt; 
#&gt; 
#&gt; 
#&gt; 
#&gt; 
#&gt; [[2]]
#&gt; [[2]]$snp
#&gt; [[2]]$snp$name
#&gt; [1] &quot;rs9939609&quot;
#&gt; 
#&gt; [[2]]$snp$chromosome
#&gt; [1] &quot;16&quot;
#&gt; 
#&gt; [[2]]$snp$position
#&gt; [1] &quot;53786615&quot;
#&gt; 
#&gt; 
#&gt; [[2]]$user
#&gt; [[2]]$user$name
#&gt; [1] &quot;Senficon&quot;
#&gt; 
#&gt; [[2]]$user$id
#&gt; [1] 2
#&gt; 
#&gt; [[2]]$user$genotypes
#&gt; list()
</code></pre>
<h3 id="phenotype-user-data">Phenotype user data</h3>
<p>Get phenotype data for one or multiple users</p>
<pre><code class="language-r">phenotypes(userid=1)$phenotypes[1:3]
#&gt; $`Caffeine dependence`
#&gt; $`Caffeine dependence`$phenotype_id
#&gt; [1] 538
#&gt; 
#&gt; $`Caffeine dependence`$variation
#&gt; [1] &quot;No&quot;
#&gt; 
#&gt; 
#&gt; $`hair on ear`
#&gt; $`hair on ear`$phenotype_id
#&gt; [1] 254
#&gt; 
#&gt; $`hair on ear`$variation
#&gt; [1] &quot;No&quot;
#&gt; 
#&gt; 
#&gt; $`Third Nipple`
#&gt; $`Third Nipple`$phenotype_id
#&gt; [1] 259
#&gt; 
#&gt; $`Third Nipple`$variation
#&gt; [1] &quot;None&quot;
</code></pre>
<pre><code class="language-r">phenotypes(userid='1,6,8', df=TRUE)[[1]][1:10,]
#&gt; Error: Bad Gateway (HTTP 502)
</code></pre>
<pre><code class="language-r">out &lt;- phenotypes(userid='1-8', df=TRUE)
lapply(out, head)
#&gt; $`Bastian Greshake Tzovaras`
#&gt;                               phenotype phenotypeID variation
#&gt; 1                   Caffeine dependence         538        No
#&gt; 2                           hair on ear         254        No
#&gt; 3                          Third Nipple         259      None
#&gt; 4                            Alcoholism         485      None
#&gt; 5        Alcohol Consumption (per week)         484         0
#&gt; 6 Allergy to artificial grape flavoring         352        No
#&gt; 
#&gt; $Senficon
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt; 
#&gt; $`no info on user_3`
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt; 
#&gt; $`no info on user_4`
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt; 
#&gt; $`no info on user_5`
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt; 
#&gt; $`Nash Parovoz`
#&gt;                          phenotype phenotypeID        variation
#&gt; 1         Y-DNA Haplogroup (ISOGG)         150        J-FGC5206
#&gt; 2  The Dress: Perception of colour         338   White and gold
#&gt; 3           Number of wisdom teeth          57                4
#&gt; 4 Ability to find a bug in openSNP           5   extremely high
#&gt; 5              Lactose intolerance           2 lactose-tolerant
#&gt; 6                       white skin           4        Caucasian
#&gt; 
#&gt; $`no info on user_7`
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt; 
#&gt; $`Samantha B. Clark`
#&gt;                             phenotype phenotypeID           variation
#&gt; 1                            Gambling         539                  No
#&gt; 2                 Caffeine dependence         538                  No
#&gt; 3            Dietary supplements used         534                 b12
#&gt; 4                                Diet         533 Vegan / plant-based
#&gt; 5                   Tooth sensitivity         532         Sweet, cold
#&gt; 6 OCD - Obsessive-Compulsive Disorder         555                  No
</code></pre>
<h3 id="all-known-variations">All known variations</h3>
<p>Get all known variations and all users sharing that phenotype for one phenotype(-ID).</p>
<pre><code class="language-r">phenotypes_byid(phenotypeid=12, return_ = 'desc')
#&gt; $id
#&gt; [1] 12
#&gt; 
#&gt; $characteristic
#&gt; [1] &quot;Beard Color&quot;
#&gt; 
#&gt; $description
#&gt; [1] &quot;coloration of facial hair&quot;
</code></pre>
<pre><code class="language-r">phenotypes_byid(phenotypeid=12, return_ = 'knownvars')
#&gt; $known_variations
#&gt; $known_variations[[1]]
#&gt; [1] &quot;Red&quot;
#&gt; 
#&gt; $known_variations[[2]]
#&gt; [1] &quot;Blonde&quot;
#&gt; 
#&gt; $known_variations[[3]]
#&gt; [1] &quot;Red-brown&quot;
#&gt; 
#&gt; $known_variations[[4]]
#&gt; [1] &quot;Red-blonde-brown-black(in diferent parts i have different color,for example near the lips blond-red&quot;
#&gt; 
#&gt; $known_variations[[5]]
#&gt; [1] &quot;No beard-female&quot;
#&gt; 
#&gt; $known_variations[[6]]
#&gt; [1] &quot;Brown-black&quot;
#&gt; 
#&gt; $known_variations[[7]]
#&gt; [1] &quot;Blonde-brown&quot;
#&gt; 
#&gt; $known_variations[[8]]
#&gt; [1] &quot;Black&quot;
#&gt; 
#&gt; $known_variations[[9]]
#&gt; [1] &quot;Dark brown with minor blondish-red&quot;
#&gt; 
#&gt; $known_variations[[10]]
#&gt; [1] &quot;Brown-grey&quot;
#&gt; 
#&gt; $known_variations[[11]]
#&gt; [1] &quot;Red-blonde-brown-black&quot;
#&gt; 
#&gt; $known_variations[[12]]
#&gt; [1] &quot;Blond-brown&quot;
#&gt; 
#&gt; $known_variations[[13]]
#&gt; [1] &quot;Brown, some red&quot;
#&gt; 
#&gt; $known_variations[[14]]
#&gt; [1] &quot;Brown&quot;
#&gt; 
#&gt; $known_variations[[15]]
#&gt; [1] &quot;Brown-gray&quot;
#&gt; 
#&gt; $known_variations[[16]]
#&gt; [1] &quot;Never had a beard&quot;
#&gt; 
#&gt; $known_variations[[17]]
#&gt; [1] &quot;I'm a woman&quot;
#&gt; 
#&gt; $known_variations[[18]]
#&gt; [1] &quot;Black-brown-blonde&quot;
#&gt; 
#&gt; $known_variations[[19]]
#&gt; [1] &quot;Was red-brown now mixed with gray,&quot;
#&gt; 
#&gt; $known_variations[[20]]
#&gt; [1] &quot;Red-blonde-brown&quot;
#&gt; 
#&gt; $known_variations[[21]]
#&gt; [1] &quot;Dark brown w/few blonde &amp; red hairs&quot;
#&gt; 
#&gt; $known_variations[[22]]
#&gt; [1] &quot;Dark blonde with red and light blonde on goatee area.&quot;
#&gt; 
#&gt; $known_variations[[23]]
#&gt; [1] &quot;Black with few red hairs&quot;
#&gt; 
#&gt; $known_variations[[24]]
#&gt; [1] &quot;Black, graying&quot;
#&gt; 
#&gt; $known_variations[[25]]
#&gt; [1] &quot;Red, moustache still is, beard mostly white&quot;
#&gt; 
#&gt; $known_variations[[26]]
#&gt; [1] &quot;Blonde/brown-some black-and red on chin-all starting to gray&quot;
#&gt; 
#&gt; $known_variations[[27]]
#&gt; [1] &quot;Dark brown&quot;
#&gt; 
#&gt; $known_variations[[28]]
#&gt; [1] &quot;Every possible color, most hair shafts have more than one color at different points along the shaft&quot;
#&gt; 
#&gt; $known_variations[[29]]
#&gt; [1] &quot;Black with few white hairs&quot;
#&gt; 
#&gt; $known_variations[[30]]
#&gt; [1] &quot;Brown ginger&quot;
#&gt; 
#&gt; $known_variations[[31]]
#&gt; [1] &quot;Dark blonde&quot;
#&gt; 
#&gt; $known_variations[[32]]
#&gt; [1] &quot;Black - going white due to age&quot;
#&gt; 
#&gt; $known_variations[[33]]
#&gt; [1] &quot;N/a&quot;
</code></pre>
<pre><code class="language-r">phenotypes_byid(phenotypeid=12, return_ = 'users')[1:10,]
#&gt;    user_id                                                                                           variation
#&gt; 1       22                                                                                                 Red
#&gt; 2        1                                                                                              Blonde
#&gt; 3       26                                                                                           red-brown
#&gt; 4       10 Red-Blonde-Brown-Black(in diferent parts i have different color,for example near the lips blond-red
#&gt; 5       14                                                                                     No beard-female
#&gt; 6       42                                                                                         Brown-black
#&gt; 7       45 Red-Blonde-Brown-Black(in diferent parts i have different color,for example near the lips blond-red
#&gt; 8       16                                                                                        blonde-brown
#&gt; 9        8                                                                                     No beard-female
#&gt; 10     661                                                                                         Brown-black
</code></pre>
<h2 id="ncbi-snp-data">NCBI SNP data</h2>
<h3 id="dbsnp">dbSNP</h3>
<p>Query NCBI’s <a href="https://www.ncbi.nlm.nih.gov/snp/">dbSNP</a> for information on a set of SNPs.</p>
<p>An example with four markers, where one has been merged, and one has been withdrawn from NCBI.</p>
<pre><code class="language-r">snps &lt;- c(&quot;rs332&quot;, &quot;rs420358&quot;, &quot;rs1837253&quot;, &quot;rs1209415715&quot;, &quot;rs111068718&quot;)
(dbsnp_info &lt;- ncbi_snp_query(snps))
#&gt; # A tibble: 4 × 16
#&gt;   query        chromosome       bp class rsid  gene  alleles ancestral_allele variation_allele seqname hgvs  assembly ref_seq minor    maf maf_population
#&gt;   &lt;chr&gt;        &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;   &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt;   &lt;chr&gt;  &lt;dbl&gt; &lt;list&gt;        
#&gt; 1 rs332        7            1.18e8 del   rs12… &quot;CFT… TTT, d… TTT              delTTT           NC_000… NC_0… GRCh38.… &lt;NA&gt;    &lt;NA&gt;  NA     &lt;df [1 × 4]&gt;  
#&gt; 2 rs420358     1            4.03e7 snv   rs42… &quot;&quot;    A,C,G,T A                C,G,T            NC_000… NC_0… GRCh38.… &lt;NA&gt;    &lt;NA&gt;  NA     &lt;df [19 × 4]&gt; 
#&gt; 3 rs1837253    5            1.11e8 snv   rs18… &quot;&quot;    T,C     T                C                NC_000… NC_0… GRCh38.… T       C      0.726 &lt;df [26 × 4]&gt; 
#&gt; 4 rs1209415715 9            4.18e7 snv   rs12… &quot;&quot;    T,A,C   T                A,C              NC_000… NC_0… GRCh38.… &lt;NA&gt;    &lt;NA&gt;  NA     &lt;df [1 × 4]&gt;
</code></pre>
<p>The maf column contains the minor allele frequency from the GnomAD database (if available). All population specific allele frequencies can be accessed through the column <code>maf_population</code> which returns a list.</p>
<pre><code class="language-r">dbsnp_info$maf_population
#&gt; [[1]]
#&gt;   study ref_seq Minor MAF
#&gt; 1            NA    NA  NA
#&gt; 
#&gt; [[2]]
#&gt;             study ref_seq Minor       MAF
#&gt; 1     1000Genomes       A     C 0.9261399
#&gt; 2          ALSPAC       A     C 0.8227815
#&gt; 3        Estonian       A     C 0.7895089
#&gt; 4       GENOME_DK       A     C 0.8750000
#&gt; 5            GoNL       A     C 0.8266533
#&gt; 6          KOREAN       A     C 0.9658703
#&gt; 7  NorthernSweden       A     C 0.8183333
#&gt; 8          Qatari       A     C 0.8379630
#&gt; 9        SGDP_PRJ       A     C 0.9175824
#&gt; 10       Siberian       A     C 0.8333333
#&gt; 11          TOMMO       A     C 0.9589499
#&gt; 12          TOMMO       A     C 0.9600821
#&gt; 13         TOPMED       A     C 0.8767313
#&gt; 14        TWINSUK       A     C 0.8193096
#&gt; 15     Vietnamese       A     C 0.9952830
#&gt; 16  dbGaP_PopFreq       A     C 0.7991653
#&gt; 17         KOREAN       A     G 0.0000000
#&gt; 18         KOREAN       A     T 0.0000000
#&gt; 19  dbGaP_PopFreq       A     T 0.0000000
#&gt; 
#&gt; [[3]]
#&gt;             study ref_seq Minor       MAF
#&gt; 1     1000Genomes       T     C 0.6178115
#&gt; 2     1000Genomes       T     C 0.6238289
#&gt; 3          ALSPAC       T     C 0.7477945
#&gt; 4       Daghestan       T     C 0.6856128
#&gt; 5        Estonian       T     C 0.7037946
#&gt; 6     GENOGRAPHIC       T     C 0.7239609
#&gt; 7       GENOME_DK       T     C 0.7250000
#&gt; 8          GnomAD       T     C 0.7257767
#&gt; 9            GoNL       T     C 0.7274549
#&gt; 10  HGDP_Stanford       T     C 0.6602687
#&gt; 11         HapMap       T     C 0.6054025
#&gt; 12         KOREAN       T     C 0.3969283
#&gt; 13        Korea1K       T     C 0.3733624
#&gt; 14 NorthernSweden       T     C 0.6850000
#&gt; 15     PAGE_STUDY       T     C 0.6673868
#&gt; 16     PRJEB36033       T     C 1.0000000
#&gt; 17     PRJEB37584       T     C 0.4141414
#&gt; 18         Qatari       T     C 0.7824074
#&gt; 19       SGDP_PRJ       T     C 0.7670940
#&gt; 20       Siberian       T     C 0.7826087
#&gt; 21          TOMMO       T     C 0.3405728
#&gt; 22          TOMMO       T     C 0.3406469
#&gt; 23         TOPMED       T     C 0.7196758
#&gt; 24        TWINSUK       T     C 0.7437972
#&gt; 25     Vietnamese       T     C 0.4074074
#&gt; 26  dbGaP_PopFreq       T     C 0.7300999
#&gt; 
#&gt; [[4]]
#&gt;           study ref_seq Minor MAF
#&gt; 1 dbGaP_PopFreq       T     A   0
</code></pre>
</div>
<div class="include-after">
</div>
<script src="https://cdn.jsdelivr.net/npm/prismjs@1.29.0/components/prism-core.min.js" defer></script>
<script src="https://cdn.jsdelivr.net/npm/prismjs@1.29.0/plugins/autoloader/prism-autoloader.min.js" defer></script>
</body>
</html>
