<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title>rsnps tutorial</title>

<script type="text/javascript">
window.onload = function() {
  var imgs = document.getElementsByTagName('img'), i, img;
  for (i = 0; i < imgs.length; i++) {
    img = imgs[i];
    // center an image if it is the only element of its parent
    if (img.parentElement.childElementCount === 1)
      img.parentElement.style.textAlign = 'center';
  }
};
</script>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: #990073
   }

   pre .number {
     color: #099;
   }

   pre .comment {
     color: #998;
     font-style: italic
   }

   pre .keyword {
     color: #900;
     font-weight: bold
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: #d14;
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>



<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 13px;
}

body {
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 20px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 {
   font-size:2.2em;
}

h2 {
   font-size:1.8em;
}

h3 {
   font-size:1.4em;
}

h4 {
   font-size:1.0em;
}

h5 {
   font-size:0.9em;
}

h6 {
   font-size:0.8em;
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre, img {
  max-width: 100%;
}
pre {
  overflow-x: auto;
}
pre code {
   display: block; padding: 0.5em;
}

code {
  font-size: 92%;
  border: 1px solid #ccc;
}

code[class] {
  background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * {
      background: transparent !important;
      color: black !important;
      filter:none !important;
      -ms-filter: none !important;
   }

   body {
      font-size:12pt;
      max-width:100%;
   }

   a, a:visited {
      text-decoration: underline;
   }

   hr {
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote {
      padding-right: 1em;
      page-break-inside: avoid;
   }

   tr, img {
      page-break-inside: avoid;
   }

   img {
      max-width: 100% !important;
   }

   @page :left {
      margin: 15mm 20mm 15mm 10mm;
   }

   @page :right {
      margin: 15mm 10mm 15mm 20mm;
   }

   p, h2, h3 {
      orphans: 3; widows: 3;
   }

   h2, h3 {
      page-break-after: avoid;
   }
}
</style>



</head>

<body>
<!--
%\VignetteEngine{knitr::knitr}
%\VignetteIndexEntry{rsnps tutorial}
-->

<h1>rsnps tutorial</h1>

<h2>Install and load library</h2>

<p>When available on CRAN</p>

<pre><code class="r">install.packages(&quot;rsnps&quot;)
</code></pre>

<p>Or get from Github</p>

<pre><code class="r">install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;ropensci/rsnps&quot;)
</code></pre>

<pre><code class="r">library(rsnps)
</code></pre>

<h2>OpenSNP data</h2>

<h3>All Genotypes</h3>

<p>Get genotype data for all users at a particular SNP</p>

<pre><code class="r">allgensnp(snp=&#39;rs7412&#39;)[1:3]
#&gt; [[1]]
#&gt; [[1]]$snp
#&gt; [[1]]$snp$name
#&gt; [1] &quot;rs7412&quot;
#&gt; 
#&gt; [[1]]$snp$chromosome
#&gt; [1] &quot;19&quot;
#&gt; 
#&gt; [[1]]$snp$position
#&gt; [1] &quot;44908822&quot;
#&gt; 
#&gt; 
#&gt; [[1]]$user
#&gt; [[1]]$user$name
#&gt; [1] &quot;R.M. Holston&quot;
#&gt; 
#&gt; [[1]]$user$id
#&gt; [1] 22
#&gt; 
#&gt; [[1]]$user$genotypes
#&gt; [[1]]$user$genotypes[[1]]
#&gt; [[1]]$user$genotypes[[1]]$genotype_id
#&gt; [1] 8
#&gt; 
#&gt; [[1]]$user$genotypes[[1]]$local_genotype
#&gt; [1] &quot;CC&quot;
#&gt; 
#&gt; 
#&gt; 
#&gt; 
#&gt; 
#&gt; [[2]]
#&gt; [[2]]$snp
#&gt; [[2]]$snp$name
#&gt; [1] &quot;rs7412&quot;
#&gt; 
#&gt; [[2]]$snp$chromosome
#&gt; [1] &quot;19&quot;
#&gt; 
#&gt; [[2]]$snp$position
#&gt; [1] &quot;44908822&quot;
#&gt; 
#&gt; 
#&gt; [[2]]$user
#&gt; [[2]]$user$name
#&gt; [1] &quot;Mom to AG&quot;
#&gt; 
#&gt; [[2]]$user$id
#&gt; [1] 387
#&gt; 
#&gt; [[2]]$user$genotypes
#&gt; [[2]]$user$genotypes[[1]]
#&gt; [[2]]$user$genotypes[[1]]$genotype_id
#&gt; [1] 173
#&gt; 
#&gt; [[2]]$user$genotypes[[1]]$local_genotype
#&gt; [1] &quot;CC&quot;
#&gt; 
#&gt; 
#&gt; 
#&gt; 
#&gt; 
#&gt; [[3]]
#&gt; [[3]]$snp
#&gt; [[3]]$snp$name
#&gt; [1] &quot;rs7412&quot;
#&gt; 
#&gt; [[3]]$snp$chromosome
#&gt; [1] &quot;19&quot;
#&gt; 
#&gt; [[3]]$snp$position
#&gt; [1] &quot;44908822&quot;
#&gt; 
#&gt; 
#&gt; [[3]]$user
#&gt; [[3]]$user$name
#&gt; [1] &quot;Dan Bolser&quot;
#&gt; 
#&gt; [[3]]$user$id
#&gt; [1] 254
#&gt; 
#&gt; [[3]]$user$genotypes
#&gt; list()
</code></pre>

<pre><code class="r">allgensnp(&#39;rs7412&#39;, df=TRUE)[1:10,]
#&gt;    snp_name snp_chromosome snp_position         user_name user_id genotype_id genotype   NA   NA   NA
#&gt; 1    rs7412             19     44908822      R.M. Holston      22           8       CC &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 2    rs7412             19     44908822         Mom to AG     387         173       CC &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 3    rs7412             19     44908822        Dan Bolser     254        &lt;NA&gt;     &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 4    rs7412             19     44908822                Lb      14           6       CC &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 5    rs7412             19     44908822 Glenn Allen Nolen      19           7       CC &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 6    rs7412             19     44908822          kevinmcc     285         118       CC &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 7    rs7412             19     44908822            Sigrid     569         260       CC &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 8    rs7412             19     44908822        Razib Khan      33          12       CT &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 9    rs7412             19     44908822             sagan      13           4       CC &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 10   rs7412             19     44908822   William Vencill     581         266       CC &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt;      NA   NA   NA   NA   NA   NA   NA   NA
#&gt; 1  &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 2  &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 3  &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 4  &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 5  &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 6  &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 7  &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 8  &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 9  &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
#&gt; 10 &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
</code></pre>

<h3>All Phenotypes</h3>

<p>Get all phenotypes, their variations, and how many users have data available for a given phenotype</p>

<p>Get all data</p>

<pre><code class="r">allphenotypes(df = TRUE)[1:10,]
#&gt;    id characteristic                   known_variations number_of_users
#&gt; 1   1      Eye color                              Brown             672
#&gt; 2   1      Eye color                        Brown-green             672
#&gt; 3   1      Eye color                         Blue-green             672
#&gt; 4   1      Eye color                          Blue-grey             672
#&gt; 5   1      Eye color                              Green             672
#&gt; 6   1      Eye color                               Blue             672
#&gt; 7   1      Eye color                              Hazel             672
#&gt; 8   1      Eye color                              Mixed             672
#&gt; 9   1      Eye color                          Gray-blue             672
#&gt; 10  1      Eye color Blue-grey; broken amber collarette             672
</code></pre>

<p>Output a list, then call the characterisitc of interest by &#39;id&#39; or &#39;characteristic&#39;</p>

<pre><code class="r">datalist &lt;- allphenotypes()
</code></pre>

<p>Get a list of all characteristics you can call</p>

<pre><code class="r">names(datalist)[1:10]
#&gt;  [1] &quot;Eye color&quot;           &quot;Handedness&quot;          &quot;Height&quot;              &quot;Sex&quot;                
#&gt;  [5] &quot;Hair Color&quot;          &quot;Tongue roller&quot;       &quot;Colour Blindness&quot;    &quot;Hair Type&quot;          
#&gt;  [9] &quot;Lactose intolerance&quot; &quot;Astigmatism&quot;
</code></pre>

<p>Get data.frame for <em>ADHD</em></p>

<pre><code class="r">datalist[[&quot;ADHD&quot;]]
#&gt;   id characteristic                            known_variations number_of_users
#&gt; 1 29           ADHD                                       False             167
#&gt; 2 29           ADHD                                        True             167
#&gt; 3 29           ADHD              Undiagnosed, but probably true             167
#&gt; 4 29           ADHD                                          No             167
#&gt; 5 29           ADHD                                         Yes             167
#&gt; 6 29           ADHD                               Not diagnosed             167
#&gt; 7 29           ADHD Diagnosed as not having but with some signs             167
#&gt; 8 29           ADHD                                 Mthfr c677t             167
#&gt; 9 29           ADHD                                   Rs1801260             167
</code></pre>

<p>Get data.frame for <em>mouth size</em> and <em>SAT Writing</em></p>

<pre><code class="r">datalist[c(&quot;mouth size&quot;,&quot;SAT Writing&quot;)]
#&gt; $`mouth size`
#&gt;    id characteristic known_variations number_of_users
#&gt; 1 120     mouth size           Medium              98
#&gt; 2 120     mouth size            Small              98
#&gt; 3 120     mouth size            Large              98
#&gt; 
#&gt; $`SAT Writing`
#&gt;    id characteristic                                        known_variations number_of_users
#&gt; 1  41    SAT Writing                                                     750              66
#&gt; 2  41    SAT Writing                                      Tested before 2005              66
#&gt; 3  41    SAT Writing                                                     800              66
#&gt; 4  41    SAT Writing                                     Country with no sat              66
#&gt; 5  41    SAT Writing                                                     N/a              66
#&gt; 6  41    SAT Writing                                 Never &amp; have ba &amp; above              66
#&gt; 7  41    SAT Writing                                                     720              66
#&gt; 8  41    SAT Writing                         Did well - don&#39;t remember score              66
#&gt; 9  41    SAT Writing                                                     511              66
#&gt; 10 41    SAT Writing                                                     700              66
#&gt; 11 41    SAT Writing                                                     760              66
#&gt; 12 41    SAT Writing                                                     780              66
#&gt; 13 41    SAT Writing Not part of sat when i took test in august 1967 at uiuc              66
</code></pre>

<h3>Annotations</h3>

<p>Get just the metadata</p>

<pre><code class="r">annotations(snp = &#39;rs7903146&#39;, output = &#39;metadata&#39;)
#&gt;          .id        V1
#&gt; 1       name rs7903146
#&gt; 2 chromosome        10
#&gt; 3   position 112998590
</code></pre>

<p>Just from PLOS journals</p>

<pre><code class="r">annotations(snp = &#39;rs7903146&#39;, output = &#39;plos&#39;)[c(1:2),]
#&gt;                author
#&gt; 1 Marguerite R. Irvin
#&gt; 2        Huixiao Hong
#&gt;                                                                                                                                            title
#&gt; 1 Genome-Wide Detection of Allele Specific Copy Number Variation Associated with Insulin Resistance in African Americans from the HyperGEN Study
#&gt; 2                                                     Technical Reproducibility of Genotyping SNP Arrays Used in Genome-Wide Association Studies
#&gt;       publication_date number_of_readers                                            url
#&gt; 1 2011-08-25T00:00:00Z              2509 http://dx.doi.org/10.1371/journal.pone.0024052
#&gt; 2 2012-09-07T00:00:00Z              3052 http://dx.doi.org/10.1371/journal.pone.0044483
#&gt;                            doi
#&gt; 1 10.1371/journal.pone.0024052
#&gt; 2 10.1371/journal.pone.0044483
</code></pre>

<p>Just from SNPedia</p>

<pre><code class="r">annotations(snp = &#39;rs7903146&#39;, output = &#39;snpedia&#39;)
#&gt;                                               url
#&gt; 1 http://www.snpedia.com/index.php/Rs7903146(C;C)
#&gt; 2 http://www.snpedia.com/index.php/Rs7903146(C;T)
#&gt; 3 http://www.snpedia.com/index.php/Rs7903146(T;T)
#&gt;                                                            summary
#&gt; 1 Normal (lower) risk of Type 2 Diabetes and Gestational Diabetes.
#&gt; 2     1.4x increased risk for diabetes (and perhaps colon cancer).
#&gt; 3                            2x increased risk for Type-2 diabetes
</code></pre>

<p>Get all annotations</p>

<pre><code class="r">annotations(snp = &#39;rs7903146&#39;, output = &#39;all&#39;)[1:5,]
#&gt;        .id                      author
#&gt; 1 mendeley        Dhanasekaran Bodhini
#&gt; 2 mendeley Ludmila Alves Sanches Dutra
#&gt; 3 mendeley               Thomas Hansen
#&gt; 4 mendeley    Laura J Rasmussen-Torvik
#&gt; 5 mendeley                      Yu Yan
#&gt;                                                                                                                                                                           title
#&gt; 1                                        The rs12255372(G/T) and rs7903146(C/T) polymorphisms of the TCF7L2 gene are associated with type 2 diabetes mellitus in Asian Indians.
#&gt; 2                                                            Allele-specific PCR assay to genotype SNP rs7903146 in TCF7L2 gene for rapid screening of diabetes susceptibility.
#&gt; 3                                                                                               At-Risk Variant in TCF7L2 for Type II Diabetes Increases Risk of Schizophrenia.
#&gt; 4                                           Preliminary report: No association between TCF7L2 rs7903146 and euglycemic-clamp-derived insulin sensitivity in a mixed-age cohort.
#&gt; 5 The transcription factor 7-like 2 (TCF7L2) polymorphism may be associated with focal arteriolar narrowing in Caucasians with hypertension or without diabetes: the ARIC Study
#&gt;   publication_year number_of_readers open_access
#&gt; 1             2007                 8       FALSE
#&gt; 2             2008                 5       FALSE
#&gt; 3             2011                 1       FALSE
#&gt; 4             2009                 3       FALSE
#&gt; 5             2010                 5        TRUE
#&gt;                                                                                                                                                                      url
#&gt; 1                               http://www.mendeley.com/research/rs12255372-g-t-rs7903146-c-t-polymorphisms-tcf7l2-gene-associated-type-2-diabetes-mellitus-asian-ind-1/
#&gt; 2                     http://www.mendeley.com/research/allelespecific-pcr-assay-to-genotype-snp-rs7903146-in-tcf7l2-gene-for-rapid-screening-of-diabetes-susceptibility/
#&gt; 3                                                                  http://www.mendeley.com/research/atrisk-variant-tcf7l2-type-ii-diabetes-increases-risk-schizophrenia/
#&gt; 4                   http://www.mendeley.com/research/preliminary-report-association-between-tcf7l2-rs7903146-euglycemicclampderived-insulin-sensitivity-mixedage-cohort/
#&gt; 5 http://www.mendeley.com/research/transcription-factor-7like-2-tcf7l2-polymorphism-associated-focal-arteriolar-narrowing-caucasians-hypertension-diabetes-aric-study-7/
#&gt;                              doi publication_date summary first_author pubmed_link journal trait pvalue
#&gt; 1                           none             &lt;NA&gt;    &lt;NA&gt;         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA
#&gt; 2                           none             &lt;NA&gt;    &lt;NA&gt;         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA
#&gt; 3 10.1016/j.biopsych.2011.01.031             &lt;NA&gt;    &lt;NA&gt;         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA
#&gt; 4                           none             &lt;NA&gt;    &lt;NA&gt;         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA
#&gt; 5         10.1186/1472-6823-10-9             &lt;NA&gt;    &lt;NA&gt;         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA
#&gt;   pvalue_description confidence_interval
#&gt; 1               &lt;NA&gt;                &lt;NA&gt;
#&gt; 2               &lt;NA&gt;                &lt;NA&gt;
#&gt; 3               &lt;NA&gt;                &lt;NA&gt;
#&gt; 4               &lt;NA&gt;                &lt;NA&gt;
#&gt; 5               &lt;NA&gt;                &lt;NA&gt;
</code></pre>

<h3>Download</h3>

<p>Download genotype data for a user from 23andme or other repo. (not evaluated in this example)</p>

<pre><code class="r">data &lt;- users(df=TRUE)
head( data[[1]] )
fetch_genotypes(url = data[[1]][1,&quot;genotypes.download_url&quot;], rows=15)
</code></pre>

<h3>Genotype user data</h3>

<p>Genotype data for one or multiple users</p>

<pre><code class="r">genotypes(snp=&#39;rs9939609&#39;, userid=1)
#&gt; $snp
#&gt; $snp$name
#&gt; [1] &quot;rs9939609&quot;
#&gt; 
#&gt; $snp$chromosome
#&gt; [1] &quot;16&quot;
#&gt; 
#&gt; $snp$position
#&gt; [1] &quot;53786615&quot;
#&gt; 
#&gt; 
#&gt; $user
#&gt; $user$name
#&gt; [1] &quot;Bastian Greshake&quot;
#&gt; 
#&gt; $user$id
#&gt; [1] 1
#&gt; 
#&gt; $user$genotypes
#&gt; $user$genotypes[[1]]
#&gt; $user$genotypes[[1]]$genotype_id
#&gt; [1] 9
#&gt; 
#&gt; $user$genotypes[[1]]$local_genotype
#&gt; [1] &quot;AT&quot;
</code></pre>

<pre><code class="r">genotypes(&#39;rs9939609&#39;, userid=&#39;1,6,8&#39;, df=TRUE)
#&gt;    snp_name snp_chromosome snp_position         user_name user_id genotype_id genotype
#&gt; 1 rs9939609             16     53786615  Bastian Greshake       1           9       AT
#&gt; 2 rs9939609             16     53786615      Nash Parovoz       6           5       AT
#&gt; 3 rs9939609             16     53786615 Samantha B. Clark       8           2       TT
</code></pre>

<pre><code class="r">genotypes(&#39;rs9939609&#39;, userid=&#39;1-2&#39;, df=FALSE)
#&gt; [[1]]
#&gt; [[1]]$snp
#&gt; [[1]]$snp$name
#&gt; [1] &quot;rs9939609&quot;
#&gt; 
#&gt; [[1]]$snp$chromosome
#&gt; [1] &quot;16&quot;
#&gt; 
#&gt; [[1]]$snp$position
#&gt; [1] &quot;53786615&quot;
#&gt; 
#&gt; 
#&gt; [[1]]$user
#&gt; [[1]]$user$name
#&gt; [1] &quot;Bastian Greshake&quot;
#&gt; 
#&gt; [[1]]$user$id
#&gt; [1] 1
#&gt; 
#&gt; [[1]]$user$genotypes
#&gt; [[1]]$user$genotypes[[1]]
#&gt; [[1]]$user$genotypes[[1]]$genotype_id
#&gt; [1] 9
#&gt; 
#&gt; [[1]]$user$genotypes[[1]]$local_genotype
#&gt; [1] &quot;AT&quot;
#&gt; 
#&gt; 
#&gt; 
#&gt; 
#&gt; 
#&gt; [[2]]
#&gt; [[2]]$snp
#&gt; [[2]]$snp$name
#&gt; [1] &quot;rs9939609&quot;
#&gt; 
#&gt; [[2]]$snp$chromosome
#&gt; [1] &quot;16&quot;
#&gt; 
#&gt; [[2]]$snp$position
#&gt; [1] &quot;53786615&quot;
#&gt; 
#&gt; 
#&gt; [[2]]$user
#&gt; [[2]]$user$name
#&gt; [1] &quot;Senficon&quot;
#&gt; 
#&gt; [[2]]$user$id
#&gt; [1] 2
#&gt; 
#&gt; [[2]]$user$genotypes
#&gt; list()
</code></pre>

<h3>Phenotype user data</h3>

<p>Get phenotype data for one or multiple users</p>

<pre><code class="r">phenotypes(userid=1)$phenotypes[1:3]
#&gt; $`white skin`
#&gt; $`white skin`$phenotype_id
#&gt; [1] 4
#&gt; 
#&gt; $`white skin`$variation
#&gt; [1] &quot;Caucasian&quot;
#&gt; 
#&gt; 
#&gt; $`Lactose intolerance`
#&gt; $`Lactose intolerance`$phenotype_id
#&gt; [1] 2
#&gt; 
#&gt; $`Lactose intolerance`$variation
#&gt; [1] &quot;lactose-tolerant&quot;
#&gt; 
#&gt; 
#&gt; $`Eye color`
#&gt; $`Eye color`$phenotype_id
#&gt; [1] 1
#&gt; 
#&gt; $`Eye color`$variation
#&gt; [1] &quot;blue-green&quot;
</code></pre>

<pre><code class="r">phenotypes(userid=&#39;1,6,8&#39;, df=TRUE)[[1]][1:10,]
#&gt;                     phenotype phenotypeID        variation
#&gt; 1                  white skin           4        Caucasian
#&gt; 2         Lactose intolerance           2 lactose-tolerant
#&gt; 3                   Eye color           1       blue-green
#&gt; 4                   Hair Type          16         straight
#&gt; 5                      Height          15  Tall ( &gt;180cm )
#&gt; 6              Ability to Tan          14              Yes
#&gt; 7  Short-sightedness (Myopia)          21              low
#&gt; 8                 Beard Color          12           Blonde
#&gt; 9            Colour Blindness          25            False
#&gt; 10                 Strabismus          23            False
</code></pre>

<pre><code class="r">out &lt;- phenotypes(userid=&#39;1-8&#39;, df=TRUE)
lapply(out, head)
#&gt; $`Bastian Greshake`
#&gt;             phenotype phenotypeID        variation
#&gt; 1          white skin           4        Caucasian
#&gt; 2 Lactose intolerance           2 lactose-tolerant
#&gt; 3           Eye color           1       blue-green
#&gt; 4           Hair Type          16         straight
#&gt; 5              Height          15  Tall ( &gt;180cm )
#&gt; 6      Ability to Tan          14              Yes
#&gt; 
#&gt; $Senficon
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt; 
#&gt; $`no info on user_3`
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt; 
#&gt; $`no info on user_4`
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt; 
#&gt; $`no info on user_5`
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt; 
#&gt; $`Nash Parovoz`
#&gt;                          phenotype phenotypeID        variation
#&gt; 1                       Handedness           3     right-handed
#&gt; 2                        Eye color           1            brown
#&gt; 3                       white skin           4        Caucasian
#&gt; 4              Lactose intolerance           2 lactose-tolerant
#&gt; 5 Ability to find a bug in openSNP           5   extremely high
#&gt; 6           Number of wisdom teeth          57                4
#&gt; 
#&gt; $`no info on user_7`
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt; 
#&gt; $`Samantha B. Clark`
#&gt;             phenotype phenotypeID                   variation
#&gt; 1          Handedness           3                 left-handed
#&gt; 2 Lactose intolerance           2          lactose-intolerant
#&gt; 3           Eye color           1                       Brown
#&gt; 4      Ability to Tan          14                         Yes
#&gt; 5 Nicotine dependence          20 ex-smoker, 7 cigarettes/day
#&gt; 6          Hair Color          13                       brown
</code></pre>

<h3>All known variations</h3>

<p>Get all known variations and all users sharing that phenotype for one phenotype(-ID).</p>

<pre><code class="r">phenotypes_byid(phenotypeid=12, return_ = &#39;desc&#39;)
#&gt; $id
#&gt; [1] 12
#&gt; 
#&gt; $characteristic
#&gt; [1] &quot;Beard Color&quot;
#&gt; 
#&gt; $description
#&gt; [1] &quot;coloration of facial hair&quot;
</code></pre>

<pre><code class="r">phenotypes_byid(phenotypeid=12, return_ = &#39;knownvars&#39;)
#&gt; $known_variations
#&gt; $known_variations[[1]]
#&gt; [1] &quot;Red&quot;
#&gt; 
#&gt; $known_variations[[2]]
#&gt; [1] &quot;Blonde&quot;
#&gt; 
#&gt; $known_variations[[3]]
#&gt; [1] &quot;Red-brown&quot;
#&gt; 
#&gt; $known_variations[[4]]
#&gt; [1] &quot;Red-blonde-brown-black(in diferent parts i have different color,for example near the lips blond-red&quot;
#&gt; 
#&gt; $known_variations[[5]]
#&gt; [1] &quot;No beard-female&quot;
#&gt; 
#&gt; $known_variations[[6]]
#&gt; [1] &quot;Brown-black&quot;
#&gt; 
#&gt; $known_variations[[7]]
#&gt; [1] &quot;Blonde-brown&quot;
#&gt; 
#&gt; $known_variations[[8]]
#&gt; [1] &quot;Black&quot;
#&gt; 
#&gt; $known_variations[[9]]
#&gt; [1] &quot;Dark brown with minor blondish-red&quot;
#&gt; 
#&gt; $known_variations[[10]]
#&gt; [1] &quot;Brown-grey&quot;
#&gt; 
#&gt; $known_variations[[11]]
#&gt; [1] &quot;Red-blonde-brown-black&quot;
#&gt; 
#&gt; $known_variations[[12]]
#&gt; [1] &quot;Blond-brown&quot;
#&gt; 
#&gt; $known_variations[[13]]
#&gt; [1] &quot;Brown, some red&quot;
#&gt; 
#&gt; $known_variations[[14]]
#&gt; [1] &quot;Brown&quot;
#&gt; 
#&gt; $known_variations[[15]]
#&gt; [1] &quot;Brown-gray&quot;
#&gt; 
#&gt; $known_variations[[16]]
#&gt; [1] &quot;Never had a beard&quot;
#&gt; 
#&gt; $known_variations[[17]]
#&gt; [1] &quot;I&#39;m a woman&quot;
#&gt; 
#&gt; $known_variations[[18]]
#&gt; [1] &quot;Black-brown-blonde&quot;
#&gt; 
#&gt; $known_variations[[19]]
#&gt; [1] &quot;Was red-brown now mixed with gray,&quot;
#&gt; 
#&gt; $known_variations[[20]]
#&gt; [1] &quot;Red-blonde-brown&quot;
#&gt; 
#&gt; $known_variations[[21]]
#&gt; [1] &quot;Dark brown w/few blonde &amp; red hairs&quot;
#&gt; 
#&gt; $known_variations[[22]]
#&gt; [1] &quot;Dark blonde with red and light blonde on goatee area.&quot;
#&gt; 
#&gt; $known_variations[[23]]
#&gt; [1] &quot;Black with few red hairs&quot;
</code></pre>

<pre><code class="r">phenotypes_byid(phenotypeid=12, return_ = &#39;users&#39;)[1:10,]
#&gt;    user_id
#&gt; 1       22
#&gt; 2        1
#&gt; 3       26
#&gt; 4       10
#&gt; 5       14
#&gt; 6       42
#&gt; 7       45
#&gt; 8       16
#&gt; 9        8
#&gt; 10     661
#&gt;                                                                                              variation
#&gt; 1                                                                                                  Red
#&gt; 2                                                                                               Blonde
#&gt; 3                                                                                            red-brown
#&gt; 4  Red-Blonde-Brown-Black(in diferent parts i have different color,for example near the lips blond-red
#&gt; 5                                                                                      No beard-female
#&gt; 6                                                                                          Brown-black
#&gt; 7  Red-Blonde-Brown-Black(in diferent parts i have different color,for example near the lips blond-red
#&gt; 8                                                                                         blonde-brown
#&gt; 9                                                                                      No beard-female
#&gt; 10                                                                                         Brown-black
</code></pre>

<h3>OpenSNP users</h3>

<pre><code class="r">data &lt;- users(df=FALSE)
data[1:2]
#&gt; [[1]]
#&gt; [[1]]$name
#&gt; [1] &quot;gigatwo&quot;
#&gt; 
#&gt; [[1]]$id
#&gt; [1] 31
#&gt; 
#&gt; [[1]]$genotypes
#&gt; list()
#&gt; 
#&gt; 
#&gt; [[2]]
#&gt; [[2]]$name
#&gt; [1] &quot;Anu Acharya&quot;
#&gt; 
#&gt; [[2]]$id
#&gt; [1] 385
#&gt; 
#&gt; [[2]]$genotypes
#&gt; list()
</code></pre>

<h2>NCBI SNP data</h2>

<h3>LDSearch</h3>

<p>Search for SNPs in Linkage Disequilibrium with a set of SNPs</p>

<pre><code class="r">LDSearch(&quot;rs420358&quot;)
#&gt; Querying SNAP...
#&gt; Querying NCBI for up-to-date SNP annotation information...
#&gt; Done!
#&gt; $rs420358
#&gt;        Proxy      SNP Distance RSquared DPrime GeneVariant GeneName GeneDescription Major Minor   MAF
#&gt; 4   rs420358 rs420358        0    1.000  1.000  INTERGENIC      N/A             N/A     C     A 0.167
#&gt; 5   rs442418 rs420358      122    1.000  1.000  INTERGENIC      N/A             N/A     C     T 0.167
#&gt; 8   rs718223 rs420358     1168    1.000  1.000  INTERGENIC      N/A             N/A     A     G 0.167
#&gt; 6   rs453604 rs420358     2947    1.000  1.000  INTERGENIC      N/A             N/A     A     G 0.167
#&gt; 3   rs372946 rs420358      -70    0.943  1.000  INTERGENIC      N/A             N/A     G     C 0.175
#&gt; 1 rs10889290 rs420358     3987    0.800  1.000  INTERGENIC      N/A             N/A     G     A 0.200
#&gt; 2 rs10889291 rs420358     4334    0.800  1.000  INTERGENIC      N/A             N/A     C     T 0.200
#&gt; 7  rs4660403 rs420358     7021    0.800  1.000  INTERGENIC      N/A             N/A     A     G 0.200
#&gt;   NObserved Chromosome_NCBI Marker_NCBI Class_NCBI Gene_NCBI Alleles_NCBI Major_NCBI Minor_NCBI MAF_NCBI
#&gt; 4       120               1    rs420358        snp      &lt;NA&gt;          G,T          G          T       NA
#&gt; 5       120               1    rs442418        snp      &lt;NA&gt;          A/G          A          G   0.0723
#&gt; 8       120               1    rs718223        snp      &lt;NA&gt;          A/G          A          G   0.0723
#&gt; 6       120               1    rs453604        snp      &lt;NA&gt;          A/G          A          G   0.0727
#&gt; 3       120               1    rs372946        snp      &lt;NA&gt;          C,G          C          G       NA
#&gt; 1       120               1  rs10889290        snp      &lt;NA&gt;          A/G          G          A   0.0841
#&gt; 2       120               1  rs10889291        snp      &lt;NA&gt;          C/T          C          T   0.0839
#&gt; 7       120               1   rs4660403        snp      &lt;NA&gt;          A/G          A          G   0.0827
#&gt;    BP_NCBI
#&gt; 4 40341238
#&gt; 5 40341360
#&gt; 8 40342406
#&gt; 6 40344185
#&gt; 3 40341168
#&gt; 1 40345225
#&gt; 2 40345572
#&gt; 7 40348259
</code></pre>

<h3>dbSNP</h3>

<p>Query NCBI&#39;s dbSNP for information on a set of SNPs</p>

<p>An example with both merged SNPs, non-SNV SNPs, regular SNPs, SNPs not found, microsatellite</p>

<pre><code class="r">snps &lt;- c(&quot;rs332&quot;, &quot;rs420358&quot;, &quot;rs1837253&quot;, &quot;rs1209415715&quot;, &quot;rs111068718&quot;)
NCBI_snp_query(snps)
#&gt;         Query Chromosome      Marker          Class Gene   Alleles Major Minor    MAF        BP
#&gt; 1       rs332          7 rs121909001         in-del CFTR     -/TTT  &lt;NA&gt;  &lt;NA&gt;     NA 117559592
#&gt; 2    rs420358          1    rs420358            snp &lt;NA&gt;       G,T     G     T     NA  40341238
#&gt; 3   rs1837253          5   rs1837253            snp &lt;NA&gt;       C/T     C     T 0.3822 111066173
#&gt; 4 rs111068718       &lt;NA&gt; rs111068718 microsatellite &lt;NA&gt; (GT)21/24  &lt;NA&gt;  &lt;NA&gt;     NA        NA
</code></pre>

</body>

</html>
